src/safeposix/dispatcher.rs

```diff
@@ -74,6 +74,8 @@ pub union Arg {
   ulong: u64,
   cstr: *const u8,
   cstrarr: *const *const u8,
+  void: *const c_void, 
```

Comment:
@jesings we're going to have some different fields that are basically pointers, I'm not sure we need different types for all of them?

Reply:
I mean casting them is unsafe and verbose so I was thinking that it's less hassle to have a separate type for each

Reply:
Ok, yeah I'm fine with that but we should probably eventually have this organized in a way thats not confusing if we start having a long list of arg types.